packages/grafana-ui/src/components/Logs/LogRowContextProvider.test.tsx

```diff
@@ -84,6 +87,85 @@ describe('getRowContexts', () => {
   });
 });
 
+describe('LogRowContextProvider', () => {
+  describe('when requesting longer context', () => {
+    it('can request more log lines', async () => {
+      const firstResult = new MutableDataFrame({
+        refId: 'B',
+        fields: [
+          { name: 'ts', type: FieldType.time, values: [10, 9, 8, 7, 6, 5, 4, 3, 2, 1] },
+          {
+            name: 'line',
+            type: FieldType.string,
+            values: ['10', '9', '8', '7', '6', '5', '4', '3', '2', '1'],
+            labels: {},
+          },
+          {
+            name: 'id',
+            type: FieldType.string,
+            values: ['10', '9', '8', '7', '6', '5', '4', '3', '2', '1'],
+            labels: {},
+          },
+        ],
+      });
+
+      const secondResult = new MutableDataFrame({
+        refId: 'B',
+        fields: [
+          { name: 'ts', type: FieldType.time, values: [14, 13, 12] },
+          { name: 'line', type: FieldType.string, values: ['14', '13', '12'], labels: {} },
+          { name: 'id', type: FieldType.string, values: ['14', '13', '12'], labels: {} },
+        ],
+      });
+
+      let called = false;
+      const getRowContextMock = (row: LogRowModel, options?: any): Promise<DataQueryResponse> => {
+        if (!called) {
+          called = true;
+          return Promise.resolve({ data: [firstResult] });
+        }
+        return Promise.resolve({ data: [secondResult] });
+      };
+      let wrapper: any;
+      await act(async () => {
+        wrapper = await mount(
+          <LogRowContextProvider row={row} getRowContext={getRowContextMock}>
+            {({ result, errors, hasMoreContextRows, updateLimit, limit }) => {
+              return (
```

Comment:
This is probably not the best way to test this. 

One possibility is to use jest.fn() mock as children and inspect its arguments.

Other option would be to refactor LogRowContextProvider a bit. The whole LogRowContextProvider is basically just hooks. We can refactor it into a single custom hook and then create a simple wrapper component that uses it. That way we can easily test the hook in isolation.